src/components/profile/Profile.js

```diff
@@ -0,0 +1,230 @@
+/* eslint-disable camelcase */
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+
+const Profile = () => {
+  const [userData, setUserData] = useState({});
+  const [profile, myProfile] = useState({});
+  const [error, setError] = useState({ status: false, message: '' });
+  const [success, successMessage] = useState({ status: false, message: '' });
+  const [checked, setChecked] = useState(false);
+  const getProfile = async () => {
+    try {
+      const response = await axios.get('https://mocki.io/v1/28c3b7e8-c59f-4605-9613-e7a30020a07d');
+      const results = response.data.data;
+      myProfile(results);
+    } catch (errorCatch) {
+      myProfile({});
+    }
+  };
+
+  useEffect(() => {
+    getProfile();
+  }, []);
+
+  const handleInputChange = (e) => {
+    const { name } = e.target;
+    const { value } = e.target;
+    setUserData((values) => ({ ...values, [name]: value }));
+  };
+  const handleCheck = () => {
+    setChecked(!checked);
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const postData = {
+      gender: userData.gender,
+      language: userData.language,
+      currency: userData.currency,
+      location: userData.location,
+      departement: userData.departement,
+      manager: `${checked}`,
+      birthdate: userData.date,
+    };
+
+    try {
+      const response = await axios.get('https://mocki.io/v1/28353661-265c-4b5a-8596-6004580cb099');
+      if (response.data.status === 200) {
+        successMessage({ status: true, message: response.data.data.Message });
+        getProfile();
+        return setError({});
+      }
+      return setError({ status: true, message: 'An error occured,try again!' });
+    } catch (errorResponse) {
+      console.log(errorResponse);
+      if (errorResponse.response) {
+        if (errorResponse.response.status === 406) {
+          return setError({ status: true, message: 'Not Logged in' });
+        }
+        if (errorResponse.response.status === 400) {
+          return setError({ status: true, message: errorResponse.response.data.Error });
+        }
+      } else if (errorResponse.request) {
+        return setError({ status: true, message: 'Server not responding' });
+      } else if (errorResponse.config) {
+        return setError({ status: true, message: 'Internal server error' });
+      } else {
+        return setError({ status: true, message: 'Internal server error' });
+      }
+      return setError({ status: true, message: 'Internal server error' });
+    }
```

Comment:
Don't redefine the response error messages as they are already sent by the server. Check the response message.